Packet++/src/S7commLayer.cpp

```diff
@@ -0,0 +1,131 @@
+#include "EndianPortable.h"
+#include "TpktLayer.h"
+
+#include "S7commLayer.h"
+#include "TcpLayer.h"
+#include <iostream>
+
+#include <cstring>
+#include <sstream>
+
+namespace pcpp
+{
+
+	S7commLayer::S7commLayer(uint8_t msg_type, uint16_t pdu_ref, uint16_t param_length, uint16_t data_length,
+							 uint8_t error_class, uint8_t error_code)
+	{
+		size_t headerLen;
+		if (msg_type == 0x03)
+		{
+			headerLen = sizeof(s7comm_ack_data_hdr);
+			m_DataLen = headerLen;
+			m_Data = new uint8_t[headerLen];
+			memset(m_Data, 0, headerLen);
+			s7comm_ack_data_hdr *ack_d = (s7comm_ack_data_hdr *)m_Data;
+			ack_d->protocol_id = 0x32;
+			ack_d->msg_type = msg_type;
+			ack_d->reserved = 0x0000;
+			ack_d->pdu_ref = htobe16(pdu_ref);
+			ack_d->param_length = htobe16(param_length);
+			ack_d->data_length = htobe16(data_length);
+			ack_d->error_class = error_class;
+			ack_d->error_code = error_code;
+		}
+		else
+		{
+			headerLen = sizeof(s7commhdr);
+			m_DataLen = headerLen;
+			m_Data = new uint8_t[headerLen];
+			memset(m_Data, 0, headerLen);
+			s7commhdr *s7commHdr = (s7commhdr *)m_Data;
+			s7commHdr->protocol_id = 0x32;
+			s7commHdr->msg_type = msg_type;
+			s7commHdr->reserved = 0x0000;
+			s7commHdr->pdu_ref = htobe16(pdu_ref);
+			s7commHdr->param_length = htobe16(param_length);
+			s7commHdr->data_length = htobe16(data_length);
+		}
+
+		m_Protocol = S7COMM;
+	}
+
+	std::string S7commLayer::toString() const
+	{
+		std::ostringstream str;
+		std::string error;
+		if (getMsgType() == 0x03)
+		{
+			error =
+				", error class: " + std::to_string(getErrorClass()) + ", error code: " + std::to_string(getErrorCode());
+		}
+		str << "S7comm Layer, "
+			<< "msg_type: " << std::to_string(getMsgType()) << ", pdu_ref: " << std::to_string(getPduRef())
+			<< ", param_length: " << std::to_string(getParamLength())
+			<< ", data_length: " << std::to_string(getDataLength()) << error;
+
+		return str.str();
+	}
```

Comment:
This string is a bit long... maybe it should be shorter, like:
```text
S7Comm Layer, Job Request
```

Reply:
What I meant is the string to reflect the message type which can be one of:
![image](https://github.com/seladb/PcapPlusPlus/assets/9059541/e0574ff8-df97-4812-9fd0-9296a38a4508)

Reply:
@wivien19 I think you misunderstood what I meant. I think the at string should represent the message type. For example:
- If the message is of type `0x01 Job Request` the string should be: `S7Comm Layer, Job Request`
- If the message is of type `0x02 Ack` the string should be: `S7Comm Layer, Ack`
- If the message is of type `0x03 Ack-Data` the string should be: `S7Comm Layer, Ack-Data`
- If the message is of type `0x07 Userdata` the string should be: `S7Comm Layer, Userdata`
- Else the string could be: `S7Comm Layer, Unknown message`